shared/src/types/header_map/mod.rs

```diff
@@ -66,19 +73,46 @@ impl HeaderMap {
         Self { inner }
     }
 
-    pub(crate) fn contains_key(&self, hash: &Byte32) -> bool {
+    pub fn contains_key(&self, hash: &Byte32) -> bool {
+        let _trace_timer: Option<HistogramTimer> = ckb_metrics::handle().map(|metric| {
+            metric
```

Comment:
nit: no need of type annotations and remove `use ckb_metrics::HistogramTimer;`.
maybe be more consistent to use `if let`

Reply:
OK

Reply:
if we use `if let`, then the code will be like:
```rust
        let _trace_timer = if let Some(metrics) = ckb_metrics::handle() {
            Some(
                metrics
                    .ckb_header_map_ops_duration
                    .with_label_values(&["contains_key"])
                    .start_timer(),
            )
        } else {
            None
        };
```